opensearch-project/OpenSearch-Dashboards
# [Workspace] [Data Source] feat: support workspace level default data source

### Description

Support set default data source in workspace

1.  Recover data source management when in workspace
2. Restrict workspace level default data source ui settings value extend from global when getting
3. Set default data source when creating work space and add check when editing workspace

## Screenshot


https://github.com/opensearch-project/OpenSearch-Dashboards/assets/42465692/ccf06f32-edc4-4a55-a69b-1b96f98af6ce


## Testing the changes

1. Go to home and set the default data source
2. Enter a workspace and. go to data source management, the default data source would not be extended.
3. Create workspace and assign some data sources, go to data source management and you could see a default data source would be set.
4. Edit workspace and assign/ unassign data sources, if default data source is unassigned, another default data source would be set.

## Changelog

- feat: support workspace level default data source 

### Check List
